src/cargo/ops/cargo_output_metadata.rs

```diff
@@ -184,14 +184,18 @@ fn build_resolve_graph_r(
             None => true,
         })
         .filter_map(|(dep_id, deps)| {
+            let mut dep_kinds: Vec<_> = deps.iter().map(DepKindInfo::from).collect();
+            // Duplicates may appear if the same package is used by different
+            // members of a workspace with different features selected.
+            dep_kinds.dedup();
```

Comment:
The vec needs to be sorted for dedup to work properly.

Reply:
Oops, thanks!